pennylane/tape/interfaces/autograd.py

```diff
@@ -126,7 +117,13 @@ def _execute(self, params, device):
         params = [p.item() if p.shape == tuple() else p for p in params]
         params = autograd.builtins.tuple(params)
 
+        # unwrap constant parameters
+        all_params_unwrapped = [p.numpy() if isinstance(p, np.tensor) else p for p in self._all_parameter_values]
+
+        # evaluate the tape
+        self.set_parameters(all_params_unwrapped, trainable_only=False)
         res = self.execute_device(params, device=device)
+        self.set_parameters(self._all_parameter_values, trainable_only=False)

```

Comment:
The extra list comprehension and the two new `set_parameters` might add some overhead, worth benchmarking.

Reply:
However, this is the same solution the TF and Torch interfaces currently implement